vllm/v1/attention/backends/mamba1_attn.py

```diff
@@ -26,11 +27,14 @@ class Mamba1AttentionMetadata:
     context_lens_tensor: torch.Tensor
     state_indices_tensor: torch.Tensor
     has_initial_states: torch.Tensor
+    num_prefills: int
```

Comment:
should has_initial_state be `Optional[torch.Tensor]`? (I'll fix those in Mamba2AttentionMetadata) And prefer it to be in shape `(num_prefills,)` like in mamba2.

Reply:
fix for mamba2: https://github.com/vllm-project/vllm/pull/22787

Reply:
has_initial_states_p given from Mamba1AttentionMetadata is with shape  (num_prefills,) in mamba1 as well. We renamed has_initial_states to has_initial_states_p as you did in https://github.com/vllm-project/vllm/pull/22787 .